build/ci/release.yml

```diff
@@ -1396,7 +1396,7 @@ buildvariants:
   - name: goreleaser_mongocli_snapshot
     display_name: "Packaging MongoCLI (goreleaser)"
     run_on:
-      - macos-1014-codesign
+      - rhel82-arm64-small
```

Comment:
How about x86_64, I don't think we have many ARM hosts available.

Reply:
the architectures available for the notarization services are only arm64 and amd64.

Reply:
https://wiki.corp.mongodb.com/display/BUILD/How+to+use+MacOS+notary+service

Reply:
updated

Reply:
@andreaangiolillo as mentioned amd64 is x86_64 and arm64 is aarch64, they are synonyms for this context